packages/inngest/src/components/execution/als.ts

```diff
@@ -0,0 +1,52 @@
+import { type Context } from "../../types.js";
+
+export interface AsyncContext {
+  ctx: Context.Any;
+}
+
+/**
+ * A type that represents a partial, runtime-agnostic interface of
+ * `AsyncLocalStorage`.
+ */
+type AsyncLocalStorageIsh = {
+  getStore: () => AsyncContext | undefined;
+  run: <R>(store: AsyncContext, fn: () => R) => R;
+};
+
+/**
+ * A local-only variable to store the async local storage instance.
+ */
+let als: Promise<AsyncLocalStorageIsh> | undefined;
+
+/**
+ * Retrieve the async context for the current execution.
+ */
+export const getAsyncCtx = async (): Promise<AsyncContext | undefined> => {
+  return getAsyncLocalStorage().then((als) => als.getStore());
+};
```

Comment:
This is the only piece exposed to the public API, used to retrieve the input to the current function run.